### ISSUE TITLE
Tests do not compile with flag -DDEBUG and thus default debugging flags.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-parser#1076]
Sigh, this is not my OPM day.

```
[ 81%] Building CXX object lib/eclipse/CMakeFiles/ParseKEYWORD.dir/IntegrationTests/ParseKEYWORD.cpp.o
In file included from /usr/include/boost/test/unit_test.hpp:20:0,
                 from /home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:22:
<command-line>:0:7: error: expected identifier before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
<command-line>:0:7: error: expected unqualified-id before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
In file included from /usr/include/boost/config.hpp:57:0,
                 from /usr/include/boost/algorithm/string/config.hpp:14,
                 from /usr/include/boost/algorithm/string/join.hpp:14,
                 from /home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:21:
<command-line>:0:7: error: expected unqualified-id before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
<command-line>:0:7: error: expected identifier before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
<command-line>:0:7: error: expected unqualified-id before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
<command-line>:0:7: error: expected unqualified-id before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
In file included from /usr/include/boost/test/unit_test.hpp:20:0,
                 from /home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:22:
<command-line>:0:7: error: expected unqualified-id before numeric constant
/home/mblatt/DUNE-test/opm-parser/lib/eclipse/IntegrationTests/ParseKEYWORD.cpp:38:23: note: in expansion of macro ‘DEBUG’
 BOOST_AUTO_TEST_CASE( DEBUG ) {
                       ^
lib/eclipse/CMakeFiles/ParseKEYWORD.dir/build.make:54: recipe for target 'lib/eclipse/CMakeFiles/ParseKEYWORD.dir/IntegrationTests/ParseKEYWORD.cpp.o' failed
make[2]: *** [lib/eclipse/CMakeFiles/ParseKEYWORD.dir/IntegrationTests/ParseKEYWORD.cpp.o] Error 1
CMakeFiles/Makefile2:2631: recipe for target 'lib/eclipse/CMakeFiles/ParseKEYWORD.dir/all' failed
make[1]: *** [lib/eclipse/CMakeFiles/ParseKEYWORD.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```
I am on commit 8626ec305bd290308d8cd3695d1fdb71b9f27440 and using the following cache file:
```
set(METIS_ROOT "/home/mblatt/opt" CACHE STRING "metis home")
set(PARMETIS_ROOT "/home/mblatt/opt" CACHE STRING "parmentis home")
set(Ecl "/home/mblatt/opt/libecl/share/cmake/ecl" CACHE STRING "ert home")
set(CMAKE_CXX_FLAGS "-DDEBUG -g -O0 -Wall -Wextra -DDEBUG " CACHE STRING "CXX opts")
set(CMAKE_C_FLAGS "-DDEBUG -g -O0 -Wall -Wextra -DDEBUG " CACHE STRING "CXX opts")
set(BUILD_TYPE "None" CACHE STRING "Build type")
set(USE_MPI ON         CACHE STRING "Use mpi")
set(PTSCOTCH_ROOT "/home/mblatt/opt" CACHE STRING "Path to PTSCOTCH")
set(ZOLTAN_ROOT "/home/mblatt/opt" CACHE STRING "Path to ZOLTAN")
set(BUILD_TESTING ON CACHE BOOL "Build tests")
```
Note that I set the rather generic `DEBUG` define which also seems to be used in the test. One could say that this is a rather generic define to set but it turns out that it is actually set by the build system in opm-common.

**Comment by joakim-hove** [comment]
https://github.com/OPM/opm-parser/pull/1078 ??

**Comment by blattms** [comment]
Sure, anything that renames DEBUG is fine.

**Comment by blattms** [comment]
Closed with the merge of #1078
